### PR TITLE
feat(searchbox): add prop enableAppbase

### PR DESCRIPTION
**Pull request by ShahAnuj2610**
dependent on: https://github.com/appbaseio/searchbox/pull/23

### CODE REVIEW

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 1]
```diff
@@ -75,6 +75,9 @@ Example uses of searchbox UI:
 -   **credentials** `string` [optional]
     Basic auth credentials for authentication purposes. It should be a string of the format `username:password`.
     If you are using an appbase.io app, you will find credentials under your [API credentials page](https://dashboard.appbase.io/app?view=credentials). If you are not using an appbase.io app, credentials may not be necessary - although having an open access to your Elasticsearch cluster is not recommended.
+-   **enableAppbase** `boolean`
+    enabling this prop allows you to use `reactivesearch-v3` API for making search and result requests. The advantage of this is it solves the security concern of `query injection`. It handles `query rules` and `analytics` in a better way.
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: @ShahAnuj2610 This sentence can be rephrased better.

1. It's missing the `optional` tag.
2. `enableAppbase` is `false` by default. You can set this to `true` when you're using appbase.io alongside ElasticSearch. It enables the following features:
   A. Recording of analytics events - search and clicks,
   B. Query generation happens on server side - protecting against security concerns around query injection,
   C. Apply query rules and functions for search queries,
   D. Apply additional security controls to requests: authenticate via RBAC (via JWTs) or Basic Auth, ACL based access control, IP based rate limits, IP/HTTP Referers whitelisting, fields filtering.

Comment (siddharthlatest) [reply content/docs/reactivesearch/react-searchbox/apireference.md]: Link to each step's docs within the points.

File: content/docs/reactivesearch/searchbase/overview/apireference.md [hunk 2]
```diff
@@ -357,6 +360,7 @@ These properties are automatically calculated or managed by the `Searchbase` cla
     -   **`hidden`**: `number` Total number of hidden results found
     -   **`promoted`**: `number` Total number of promoted results found
     -   **`promotedData`**: `Array<Object>` An array of promoted results obtained from the applied query.
+    -   **`customData`**: `Object` An object of custom data obtained from the `reactivesearch-v3` API.
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/searchbase/overview/apireference.md]: Whenever referencing the ReactiveSearch v3 API, link to the doc on it. This doc is different from the REST API reference doc. @bietkul We need to create this doc.

**Review by siddharthlatest — CHANGES_REQUESTED** [review]
All of the above changes would apply to searchbox docs as well.

File: content/docs/reactivesearch/react-searchbox/apireference.md (hunk 1 above)
Comment (siddharthlatest) [reply content/docs/reactivesearch/react-searchbox/apireference.md]: @ShahAnuj2610 Wouldn't the change also apply to `searchbox` and `vue-searchbox` libraries too?

File: content/docs/reactivesearch/searchbase/overview/apireference.md (hunk 2 above)
Comment (ShahAnuj2610) [reply content/docs/reactivesearch/searchbase/overview/apireference.md]: @siddharthlatest have added TODOs, will link them once the docs for the same are ready.

**Comment by ShahAnuj2610** [comment]
@siddharthlatest for `searchbox` we don't explicitly define props. You can see we haven't defined `dataField`, `nestedField`, etc... over [here](https://docs.appbase.io/docs/reactivesearch/searchbox/api/).

File: content/docs/reactivesearch/searchbase/overview/apireference.md (hunk 2 above)
Comment (siddharthlatest) [reply content/docs/reactivesearch/searchbase/overview/apireference.md]: @ShahAnuj2610 The docs are ready and live. You can now make the above change.

**Comment by siddharthlatest** [comment]
@ShahAnuj2610 You need to re-create this PR based on the current branch, and only include the specific changes.

**Comment by ShahAnuj2610** [comment]
@siddharthlatest this changes are handled here. https://github.com/appbaseio/Docs/pull/128

**Pull request closed without merging**